### ISSUE TITLE
Customizing/Renaming "Export | PDF Document"

### ISSUE THREAD

**Issue by chchen2021** [Zettlr/Zettlr#2803]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [X] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [X] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

In Assest Manager, can you offer an option to customize "Export | PDF Document"? 

This is a continuation of our earlier discussion on  [#725](https://github.com/Zettlr/Zettlr/issues/725#issuecomment-941225798):

> "PDF Document" means that Zettlr will export the file to HTML and then, using Chromium's built-in stuff "print" it to PDF.

I think you are referring to the following code:
https://github.com/Zettlr/Zettlr/tree/develop/source/main/modules/export/pdf-exporter.ts

### Example Workflow

Create a new panel similar to the "Exporting | PDF" option already existing in "Assets Manager".

### Caveats

The current "File | Export ..." options are a bit confusing. Their correspondence to the options in the Assets Manager seem to be:

| File - Export | Assets Manger - Exporting |
| ------------- | ------------------------- |
| LaTeX Source  | LaTeX                     |
| PDF Document  |                        |
| PDF (LaTeX）   | PDF                       |

Earlier versions of Zettlr might have offered an interface to customize the export to PDF Document (actually markdown to html, then to pdf). Zettlr 2 has eliminated this user interface. 

A renaming is probably needed to avoid confusion. It took me much trial and error to understand this obscure correspondence. For quite a while, I thought the "PDF Document" export option in File | Export corresponds to the "PDF" option in the Assets Manager, and "PDF (LaTeX)" corresponds to "LaTeX".

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [ ] Windows
- [X] macOS
- [ ] Linux

**Comment by nathanlesage** [comment]
So, the main problem is the following: Pandoc cannot write PDF documents on its own. So we have to trick a bit. The most common solution is to simply use a LaTeX distribution, which is what "PDF (LaTeX)" does: It prompts Pandoc to export the document to a PDF using a LaTeX distribution. However, a LaTeX distribution has to be installed on the computer. Since LaTeX distros can be quite big, and not everyone wants to install one, we also need to enable PDF exports without that.

This is facilitated via first exporting to HTML, and then opening the HTML file inside a hidden browser window and simply "printing" it to PDF using Chrome's print dialog.

That being said: The HTML export options are being used both for HTML and "PDF Document" exports. Hence, there is no specific export available for that. And I fear, adding another defaults file just for that would be overkill, since currently the exporter assumes that the defaults files are named after the writers of Pandoc, and since there is no such writer for PDF in Pandoc, this would increase the code complexity enormously, and also would lead to confusion, because the `writer`-field in the file would then be called "html", which doesn't fit with the expectation of what "PDF" means.

This is why I haven't done as you request, but I see the issue. Given this information/rationale behind my decision, what would you think might be a good course of action?

**Comment by chchen2021** [comment]
I understand the dilemma you are in. Let me offer two suggestions:

First, rename/reorder "File | Export" items to correspond with the naming **and** ordering of  "Assets Manager | Exporting". If you compare the two lists, you may appreciate why I initially made the wrong association. When I tried to customize "File | Export | PDF Document", the first PDF item on the list, I naturally looked for the first PDF item on the Assets Manger, which is simply "Assets Manager | Exporting | PDF". If Zettlr follows the pandoc convention, then I would say rename the "File | Export" list. This list also has a few other items not appearing in the Assets Manager. I do not use them so I cannot say if they are similarly confusing. You might want to consider a horizontal line dividing the "File | Export" options into two parts, the first part is customizable in the Assets Manager, while the second part is not controlled by pandoc (?).

Second, customizing the "File | Export | PDF Document" is important. By customizing, I mean simple things like paper size and margins. (The markdown rendering is fine with the HTML defaults file.) Right now, it's on an A4 default that I cannot change. The margins look weird: the horizontal margins are slightly too large, and the vertical margin are almost zero on some pages. (Not sure if the default paper size played any role in the margin.) The strange-looking margin, plus an A4 default, makes the PDF output unusable for me. Workarounds have their own problems:

- I have the option of first creating a "File | Export | HTML", and then print using my own browser, but this is quite inconvenient because the browser default has date/page numbering heading that is useful for other purposes but not desired for a markdown output. On macOS, I can also do "File | Print" from the Zettlr menu and create PDF in the printing dialog. This offers a lettersize default for me and there is no browser heading etc. However, the margin is still weird, and not accessible for customization. 

- I also have the option of using "PDF (LaTeX)", assuming the aforementioned confusion is cleared. In addition to the requirement of a LaTeX installation, there is another issue. I write bilingually in English and Chinese. The default LaTeX output setting will not print any Chinese characters. It was nontrivial to find that I need to declare a variable in the defaults file: `CJKmainfont: "Songti SC"`. The HTML export can handle bilingual files without any additional parameter, so it is more foolproof.

Not knowing the underlying architecture of Zettlr, I cannot make concrete suggestions on the user interface for customizing "PDF Document". I like the convenience of this option, but would like to be able to slightly tweak its appearance. As I said earlier, the HTML rendering of markdown is fine. What I am looking for is a simple way to quickly convert that HTML export into a usable PDF, with the right paper size and margins.

By the way, the "PDF (LaTeX)" option is very handy in creating publication-quality pdfs. In the past, I had to go out of the markdown editor to run the pandoc engine. Zettlr's export option is so much quicker.

**Comment by chchen2021** [comment]
On macOS, the papersize and margin can be controlled as follows:

1. File | Print, which brings up a new window with the html output.
2. Click the printer icon on the top right corner.
3. In the macOS print dialog, click shows details, click paper size, and then select "manage custom size" to create a custom paper size with the desired margin. (This custom profile is reusable.)
4. Save as PDF (lower left button in the print dialog) instead of actual printing. 

The resulting PDF is exactly what I was looking for. Can Zettlr offer something a one-click "PDF Document" solution, so I do not have to go through these complex steps each time.

**Comment by nathanlesage** [comment]
With regard to customizing the PDF Document settings, I think this here can come in handy: https://www.electronjs.org/docs/latest/api/webview-tag/#webviewprinttopdfoptions

I can add those options in the exporter dialog.

This still doesn't solve the confusion, though. I'm currently thinking … mh. Maybe "PDF (via HTML)"?

**Comment by chchen2021** [comment]
Yes, the webview print options have everything I need.

The naming depends a bit on how the export interface is reordered. I think "PDF (via HTML)" and "PDF (via LaTeX)" will help avoid confusion, but be consistent in the use of "via". I would also recommend abbreviating "LaTeX Source" to "LaTeX", and "RichText Format" to "RichText" to conform with the naming convention in the export interface. Calling the output of a Pandoc conversion "source" is particularly confusing. 

Like I recommended above, I think it helps to separate the export list into two categories: those directly customizable in Assets Manager, and everything else. I started using Zettlr late in the 2.0 beta phase, and found the customization of Zettlr export a bit confusing. The direct inclusion of defaults files from Pandoc is very powerful, but is also very daunting to a newcomer given the aforementioned confusions on terminology. Separating the export list will help to guide those users who need to customize the export.
